File: Program's_Contributed_By_Contributors/C++/InsertionSort.c
// C program for insertion sort
//contributed by @whyanujjwhy
#include <math.h>
#include <stdio.h>

void insertionSort(int arr[], int n)
{
	int i, j, key;
	for (i=1; i<n; i++) {
		key=arr[i];
		j=i-1;

		// Move elements of arr[0..i-1], that are
		//greater than key, to one position left
		//to their current position 
		while(j>=0 && arr[j]>key) {
			arr[j+1]=arr[j];
			j--;
		}
		arr[j+1]=key;
	}
}

// A function to print the array
void printArray(int arr[], int n) {
	int i;
	for(i=0; i<n; i++)
		printf("%d ", arr[i]);

	printf("\n");
}

/* Driver program to test insertion sort */
int main()
{
    int n;
    printf("Enter size of array: ");
    scanf("%d", &n);

    int arr[n];
    for(int i=0; i<n; i++) {
        scanf("%d", &arr[i]);
    }

	insertionSort(arr, n);
	printArray(arr, n);

	return 0;
}

File: Program's_Contributed_By_Contributors/Sorting_Algorithms/InsertionSort.java
//  implementation of Insertion Sort in java ;
import java.util.Scanner;

public class InsertionSort {
	/*Function to sort array using insertion sort*/
	void sort(int arr[])
	{
		int n = arr.length;
		for (int i = 1; i < n; ++i) {
			int key = arr[i];
			int j = i - 1;

			/* Move elements of arr[0..i-1], that are
			greater than key, to one position ahead
			of their current position */
			while (j >= 0 && arr[j] > key) {
				arr[j + 1] = arr[j];
				j = j - 1;
			}
			arr[j + 1] = key;
		}
	}

	/* A utility function to print array of size n*/
	static void printArray(int arr[])
	{
		int n = arr.length;
		for (int i = 0; i < n; ++i)
			System.out.print(arr[i] + " ");

		System.out.println();
	}

	
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);

		System.out.print("Enter the length of Array = ");
		int n = sc.nextInt();
		int arr[] = new int[n];
		System.out.print("\n Enter the values of array = ");

		for(int i =0; i<n;i++){
			System.out.print("\n Enter the value of Array at index "+ i +" = ");
			arr[i] =sc.nextInt();
		}

		InsertionSort ob = new InsertionSort();
		ob.sort(arr);

		printArray(arr);
	}
};


// This code is contributed
// by Vinayak kushwah
// #Hacktoberfest2023
